### PR TITLE
Make the build reproducible in case the SOURCE_DATE_EPOCH is known

### PR DESCRIPTION
**Pull request by yarikoptic**
To facilitate reproducible builds in Debian

See https://wiki.debian.org/ReproducibleBuilds

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [ ] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.  *Is it worth it for this one?*
- [ ] I have tested this PR locally with a `make test`
- [ ] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: libexec/bootstrap-scripts/environment/Makefile.am [hunk 1]
```diff
@@ -41,7 +41,7 @@ environment.tar:
 	ln -sf .singularity.d/actions/run newroot/.run
 	ln -sf .singularity.d/actions/shell newroot/.shell
 	ln -sf .singularity.d/actions/test newroot/.test
-	cd newroot; tar czf ../environment.tar . --owner=0 --group=0
+	cd newroot; GZIP=-n tar czf ../environment.tar . --owner=0 --group=0 --sort=name --mtime="@${SOURCE_DATE_EPOCH:-$(date +%s)}"
```
Comment (yarikoptic) [inline libexec/bootstrap-scripts/environment/Makefile.am:44]: oh, apparently `--sort` is a "recent" addition to tar ... hmm

**Review by GodloveD — COMMENTED** [review]
Hey @yarikoptic!  Sorry for the long delay on this review.  

Did you see https://github.com/singularityware/singularity/pull/1102 ?  I think it's a pretty similar PR.  @gmkurtzer suggested:
>I would like to see this check being done in configure.ac and then we export the correct tar arguments for all called tar commands (we are calling tar directly in several places).

Is that something you would like to try to do?

**Comment by yarikoptic** [comment]
nope, didn't see #1102 ...  indeed should be done in configure.ac ... if nobody does and I still remember I will do.  not sure yet which one is better, so will keep this one open for now as a reminder ;)

**Comment by c-ney** [comment]
See my comment https://github.com/singularityware/singularity/pull/1102#issuecomment-366633737.

**Comment by GodloveD** [comment]
Hey @yarikoptic and @c-ney where does this PR stand?  Is this still worth looking into?

**Comment by yarikoptic** [comment]
well, for debian builds we do patch along the lines suggested in this PR and implemented here: https://salsa.debian.org/hpc-team/singularity-container/blob/master/debian/patches/reproducible_build  

I am just waiting patiently for upstream to resolve it one way or another (or not, then we would just carry the patch forever ;) )

**Comment by jmstover** [comment]
@yarikoptic does the following Makefile output look correct to you?

```
cd newroot; GZIP=-n tar czf ../environment.tar . --owner=0 --group=0 --sort=name --mtime="@1528224051"
```

The addition of sort and mtime comes from the `./configure` script. It will only add sort and mtime if the sort option is availible in tar, otherwise it falls back to previous behavior.

**Comment by GodloveD** [comment]
Heya @jmstover thanks for having a look at this and trying to implement it at the configure stage.  Maybe it would be a good idea to open an new PR and then reference it here?

**Pull request closed without merging**